src/handlers/cloudtrail.py

```diff
@@ -0,0 +1,47 @@
+import tornado.web
+import logging
+import sys
+import gzip
+import json
+
+from src.lib.Statsd import StatsClientSingleton
+
+
+class CloudTrailHandler(tornado.web.RequestHandler):
+    """ The AWS CloudTrail handler class
+    """
+
+    def initialize(self, amqp_con):
+        """
+        handler initialisation
+        """
+        self.logger = logging.getLogger("tornado.application")
+        self.amqp_con = amqp_con
+
+    def post(self):
+        """
+        HTTP Post handler
+        * Forward request: publish to AMQP
+        :return: HTTPStatus 200
+        """
+        try:
+            StatsClientSingleton().incr('input.cloudtrail', count=1)
+            StatsClientSingleton().incr('amqp.output', count=1)
+            routing_key = self.request.uri.replace('/', '.')[1:]
+
+            payload = self.request.body
+            content_encoding = self.request.headers.get('Accept-Encoding')
+            if content_encoding == 'gzip':
```

Comment:
Do you really need the Gzip feature ?

Reply:
Yes, CloudTrail logs are gzipped by default.